activerecord/lib/active_record/database_configurations/hash_config.rb

```diff
@@ -82,7 +82,14 @@ def max_threads
       end
 
       def query_cache
-        configuration_hash[:query_cache]
+        case value = configuration_hash[:query_cache]
+        when /\A\d+\z/
+          value.to_i
+        when "false"
+          false
+        else
+          value
+        end
```

Comment:
I think this should be in the subclass, `UrlConfig`, as it doesn't make sense to do casting when provided a hash from Ruby.

Reply:
Whoops, yeah forgot we talked about this. Fixed in 6302519740.